### ISSUE TITLE
Podman Desktop fails to download components when behind transparent network proxy (self signed cert error)

### ISSUE THREAD

**Issue by failboat79** [containers/podman-desktop#9826]
### Bug description

Many corporate users will be beind a transparent proxy. This means https connections will get intercepted and the corporate cert will be inserted. Clients will fails to validate the connection with a "self-signed" cert error - unless we add the corporate cert to the root of trust. This is all very common and boring.

There is documentation for how to add the cert inside the podman machine - but I can't find how to add the cert to podman desktop itself. For this reason the program fails to download things like the Compose extension. Have tested this on multiple windows systems (both a weird lts enterprise one, and normal win11 one). 

The system I am running on does have certs configured properly in control panel. I can run "curl https://github.com" from commandline without errors - but Podman Desktop is not using the windows certs it seems. How to add them?

### Operating system

Windows 11

### Installation Method

Installer from website/GitHub releases

### Version

1.14.0

### Relevant log output

kubectl-cli] Error when downloading kubectl CLI latest release information. HttpError: self signed certificate in certificate chain
console.<computed>	@	VM5:2

**Comment by failboat79** [comment]
Some more notes: 
I tested this on other systems without corporate security software, and not on corporate network - it works fine there. 
The recent feature added to configure network proxies is not useful for solving this - as it's a transparent proxy. All traffic goes through it whether we like it or not. The only fix is to trust the corporate CA - which should be easy, but I can't see where to add it.  

edit: I haven't tested this on linux - maybe it uses the system CA properly there. On windows it definitely does not however.

**Comment by failboat79** [comment]
I manually downloaded and ran podman-compose (and added it to systemwide path). My compose applications shows up as running fine in the Desktop GUI now. But the compose extension still shows as not setup and keep trying and failing to download. Does this even matter?

**Comment by benoitf** [comment]
hello, Podman Desktop is reading the system certificates and it should be adding them to all http requests of extensions but it seems it's not so we need to check the request being involved.

Could you display a screencast or a screenshot when you have the error ?

because for example the error you reported the error `Error when downloading kubectl CLI latest release information` but this one is not related to compose but for kubectl (so at least we can check this call)

this one is delegating to https://www.npmjs.com/package/@octokit/rest

**Comment by failboat79** [comment]
There are other errors like this: 
[compose] GET /repos/docker/compose/releases - 500 with id UNKNOWN in 140ms
Error: Error invoking remote method 'command-registry:executeCommand': Error: An object could not be cloned.

My restricted environment stops me from posting videos or screenshots unfortunately.

Other posts here were blaming the error 500 on github rate limiting - but it only happens when behind transparent proxy.

**Comment by kieronlanning** [comment]
I'm in the same position as you @failboat79.

I've been using Rancher Desktop as a replacement, which works fine with my corporate proxy. However, it does fail with .NET Aspire due to a [networking bug/ support](https://github.com/dotnet/aspire/issues/6002), but that's unlikely to affect you.

**Comment by mail4taz** [comment]
I have the same problem but on MacOS Sequoia 15.2, company laptop. I can access the browser http version of the github repos for docker-compose or kubectl ok, but Podman desktop fails installing both of them.

Operating system
Sequoia 15.2

Installation Method
Installer from website https://podman-desktop.io/downloads
Tried both universal dmg and Apple Silicon dmg

Version
1.15.0

Relevant log output
main ↪️ [kind] GET /repos/kubernetes-sigs/kind/releases - 500 with id UNKNOWN in 104ms
main ↪️ [kind] Error when downloading kind CLI latest release information. Error: self signed certificate in certificate chain
main ↪️ [compose] GET /repos/docker/compose/releases - 500 with id UNKNOWN in 107ms
main ↪️ [compose] Error when loading compose latest release meta. Error: self signed certificate in certificate chain
main ↪️ [kubectl-cli] GET /repos/kubernetes/kubernetes/releases?per_page=10 - 500 with id UNKNOWN in 84ms
main ↪️ [kubectl-cli] Error when downloading kubectl CLI latest release information. HttpError: self signed certificate in certificate chain

<img width="1042" alt="Image" src="https://github.com/user-attachments/assets/695c8b41-e2c1-4233-91b3-a861bef44649" />

<img width="1042" alt="Image" src="https://github.com/user-attachments/assets/55d5557b-8fb0-4585-9da5-f6474ed7ac46" />

<img width="1042" alt="Image" src="https://github.com/user-attachments/assets/3f493518-b1bc-46c2-b08d-37cef1f5c78d" />

<img width="1042" alt="Image" src="https://github.com/user-attachments/assets/09952682-11a2-4c80-b29c-5bfc397502ad" />

**Comment by agifsnyder** [comment]
#11298 
Tested this with a very recent pre-release, which doesn't seem to have resolved the "not loading all the system (windows in my case) certs"
added more details in my comment here: https://github.com/podman-desktop/podman-desktop/issues/11298#issuecomment-2719317848

**Comment by agifsnyder** [comment]
[#](https://github.com/podman-desktop/podman-desktop/issues/11298#issuecomment-2726016363)

TL;DR; updating w/ my comment on the other thread/issue 11298 which traces back to the Windows Certificate Store having been improperly loaded with the corporate certificate chain.  That apparently caused enough of an issue when the Electron app tried loading the certificates used by my corporate network proxy.  It failed to load them properly, also failed to log that it had problems with them, and resulted in "cert errors" that were almost impossible to troubleshoot.

**Comment by kieronlanning** [comment]
Since moving back to Rancher Desktop I've had zero problems with the corporate firewall/ man-in-the-middle certs.

**Comment by benoitf** [comment]
hello, 

Podman Desktop read the certificates using this code

https://github.com/podman-desktop/podman-desktop/blob/9bbff86da2a9fe240ffecfd7fc303a2edb4fb6ce/packages/main/src/plugin/certificates.ts#L50

so, first, is that the certificates that you have on your laptop are in the files scanned there ?


then in proxy-resolver.ts and proxy.ts we override the fetch methods to use these certificates

https://github.com/podman-desktop/podman-desktop/blob/9bbff86da2a9fe240ffecfd7fc303a2edb4fb6ce/packages/main/src/plugin/proxy.ts#L203-L214

https://github.com/podman-desktop/podman-desktop/blob/9bbff86da2a9fe240ffecfd7fc303a2edb4fb6ce/packages/main/src/plugin/proxy-resolver.ts#L51

if you're able to run the development version of Podman Desktop you may be able to pin-down where the issue is as on my end I don't have a "transparent proxy" to try
